File: C++/Bubble_Sort.cpp
//? The objective of this C++ program is to implement bubble sort on an unsorted array!
#include <iostream>
using namespace std;
int main()
{
    int A[50], i, j, k, temp = 0, l;
    cout << "Enter how many elements:" << endl;
    cin >> l;
    for (i = 0; i < l; i++)
    {
        cout << "Enter element:" << endl;
        cin >> A[i];
    }
    cout << "Your Unsorted Array is:" << endl;
    for (i = 0; i < l; i++)
        cout << A[i] << endl;
    for (i = 0; i < l; i++)
    {
        for (j = 0, k = 1; j < l - 1, k < l; j++, k++)
        {
            if (A[j] >= A[k])
            {
                temp = A[j];
                A[j] = A[k];
                A[k] = temp;
            }
        }
    }
    cout << "Your Sorted Array is:" << endl;
    for (i = 0; i < l; i++)
        cout << A[i] << endl;
}

File: Python/Bubble_Sort.py
"""
todo: In this program, our objective is to develop a python code that applies buuble sort on a unsorted list.
? Bubble sort is a basic algorithm for arranging a string of numbers or other elements in the correct order. 
? The method works by examining each set of adjacent elements in the string, from left to right, switching their positions if they are out of 
? order.The algorithm then repeats this process until it can run through the entire string and find no two elements that need to be swapped.
"""
L = int(input('Please enter the size of list:'))
lst = []
i = 1
while L != 0:
    element = int(input(f'Enter Element {i} = '))
    lst.append(element)
    i += 1
    L -= 1
print('Your Unsorted List is:', lst)
no_of_pass = len(lst)
temp = 1
while(no_of_pass != 0):
    for i in range(0, len(lst)):
        for j in range(1, len(lst)):
            if lst[i-1] <= lst[j-1]:
                temp = lst[i-1]
                lst[i-1] = lst[j-1]
                lst[j-1] = temp
    no_of_pass -= 1
print('Sorted List[Using Bubble Sort]:', lst)
print('\nHit Enter to End.......')
input()
